build_scripts/ios/build.sh

```diff
@@ -1,37 +1,130 @@
+            ;;
+        t)
+            targets=($OPTARG)
+            for t in ${targets[@]}; do
+                if [[ ! " ${SUPPORTED_TARGETS[@]} " =~ " ${t} " ]]; then
+                    echo "invalid target: ${t}"
+                    echo "Supported targets are: ${SUPPORTED_TARGETS[@]}"
+                    exit 2
+                fi
+            done
+            ;;
+        *)
+            echo "unknow parameter"
+            exit 2
+            ;;
+    esac
+done
+echo "build path: ${buildpath}"
+echo "source path: ${sourcepath}"
+echo "build architectures: ${architectures[@]}"
+echo "build targets: ${targets[@]}"
+sourcepath=$(cd ${sourcepath} && pwd)   #full path
+buildpath=$(mkdir -p ${buildpath} && cd ${buildpath} && pwd)    #full path
 
-mkdir -p "${buildpath}"
-cd "${buildpath}"
-if [[ -n $(ls) ]]; then
-    echo "Error: build path '${buildpath}' not empty."
-    exit 2
-fi
-cd -
+# build framework for each architecture and target
+frameworkspath="frameworks/ios"
+for arch in ${architectures[@]}; do 
+    cd ${buildpath}
+    echo "build ${arch} framework start"
+    mkdir -p ios_build_file/${arch} && cd ios_build_file/${arch}
+    if [[ "${arch}" == "arm64" || "${arch}" == "armv7" ]]; then
+        toolchain="cmake/toolchains/ios.cmake"
+    elif [[ "${arch}" == "x86_64" || "${arch}" == "i386" ]]; then
+        toolchain="cmake/toolchains/ios_simulator.cmake"
+    fi
+    cmake -DCMAKE_TOOLCHAIN_FILE=${sourcepath}/${toolchain} \
+        -DCMAKE_OSX_ARCHITECTURES=${arch} \
+        -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=${buildpath}/${frameworkspath}/${arch} \
+        ${sourcepath}
+    cmake --build . --target ${targets[@]}
 
-set -ex
+    echo "build ${arch} framework end"
+done
+echo "${architectures[@]} frameworks build end"
+
+# arrange the framework 
+IFS=$'\n' # split $(ls) on \n characters
+cd ${buildpath}/${frameworkspath}
+for arch in ${architectures[@]}; do
+    mv ${arch}/firebase_app.framework ${arch}/firebase.framework
+    mv ${arch}/firebase.framework/firebase_app ${arch}/firebase.framework/firebase
+
+    files=$(ls ${arch})
+    for f in ${files}; do
+        if [[ ${f} =~ ".framework" ]]; then
+            rm ${arch}/${f}/Info.plist
+        else 
+            rm -rf ${arch}/${f}
+        fi
+    done
+done
+echo "${architectures[@]} frameworks ready to use"
+
+# if we built for all architectures (arm64 armv7 x86_64 i386)
+# build universal framework as well
+if [[ ${#architectures[@]} < ${#SUPPORTED_ARCHITECTURES[@]} ]]; then
+    exit 0
+fi
 
-cmake -GXcode "-DCMAKE_TOOLCHAIN_FILE=$toolchain" -S "${sourcepath}" -B "${buildpath}"
-cmake --build "${buildpath}"
+frameworks=$(ls arm64)
+for framework in ${frameworks}; do
+    mkdir -p universal/${framework}
+    libbasename="$(basename "${framework}" .framework)"
+    libsubpath="${framework}/${libbasename}"
+    if [[ -f "arm64/${libsubpath}" ]]; then
+        lipo -create "arm64/${libsubpath}" "armv7/${libsubpath}" "x86_64/${libsubpath}" "i386/${libsubpath}" \
```

Comment:
Could this list be prepared from SUPPORTED_ARCHITECTURES instead of hard coding the names?